### PR TITLE
Beckys branch

### PR DESCRIPTION
**Pull request by beckybeauchamp1**
I incorporated part of the lesson plan from `why_test.md` and changed the landlord example to take out the dependencies on active record and postgres. I would love any feedback on how I can better modify the lesson plan, i know it's too long at this point so I need to take something out as well

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,103 +1,201 @@
...
+- Write and pass unit tests using RSpec
 
-## Why learn this?
+## Framing: Introduction to Test-Driven Development
 
-We've been using the "Sinatra Reloader" gem pretty much since we first touched Sinatra.
+### Why do we implement tests in our applications? (5 min)
 
-#### Why?
+As our applications increase in complexity, we need a safety net.  We need something to ensure that we "Do no harm".  We need a battery of automated tests.  These are specifications about YOUR code that you can run to ensure your code is doing what it should.
```
Comment (jshawl) [inline readme.md:15]: to increase industry experience, recommend cloning a popular ruby project and running the test suite, like sass - there's like 1000 tests there

File: readme.md [hunk 2]
```diff
@@ -1,103 +1,201 @@
...
 
-You know what else is super-annoying? Every time you change the "POST" route of Tunr or Sinatra, even just a little bit, you have to go back to `localhost:4567/artists/10/edit` *again*, fill in the form *again*, click "submit" *again*, read the error message *again*, restart Pry *again*...
+### Unit testing (5 min)
 
-Whether or not you noticed, you're spending a **lot** of time just checking to see if your app works. That's time that could be spent adding new features, or even sleeping, Heaven forbid.
+**Unit tests** check the smallest level. The functionality of a specific method.
 
-*Life doesn't have to be this way.*
+**Acceptance tests** verify our apps at the level of user interaction.  They visit web pages, click on links, validate the DOM.
```
Comment (jshawl) [inline readme.md]: who is "they"?

File: readme.md [hunk 3]
```diff
@@ -1,103 +1,201 @@
...
+
+Let's look further into ```person_spec.rb```
+
+``` ruby
+# This first line is a reference to our library code.  We need to access to the classes we have written in Ruby to write our tests!
+
+require_relative '../models/person'  # a reference to our code
+
+describe Person do
+  describe "Constructor" do
```
Comment (jshawl) [inline readme.md:126]: or `.new`

File: readme.md [hunk 4]
```diff
@@ -1,103 +1,201 @@
...
+        # legacy syntax - the old DSL
+        tony.greeting.should eql("Ciao, mi chiamo Tony.")
+        # equivalent to:
+        # expect(tony.greeting).to eql("Ciao, mi chiamo Tony.")
       end
     end
   end
 end
 ```
+**Q. What does `expect(@matt).to be_an_instance_of(Person)` mean in regular English?**
```
Comment (jshawl) [inline readme.md:166]: a cheat sheet would be nice here for when to use `before` `let` and `subject`

File: readme.md [hunk 5]
```diff
@@ -117,69 +213,76 @@ Then, in your Terminal:
...
+$ mkdir models
+$ touch models/dog.rb
 ```
+>`rspec --init`
 
-"Spec", like "examples", basically means "tests". We're going to put some tests inside this file.
+>When you look at repos with RSpec inside them, you're likely to see two files we don't have right now: `.rspec` and `spec_helper.rb`. These files are generated by running RSpec's built-in file generator.
 
-This file requires both the `rspec` gems and the `active_record` gems.
+>Both of these files are configuration files. Neither of them are necessary unless you want to configure RSpec a certain way.
```
Comment (jshawl) [inline readme.md]: recommend adding color at least

File: readme.md [hunk 6]
```diff
@@ -228,72 +331,75 @@ You might have noticed when we ran `rspec` before that there was a little asteri
...
+
+2. We identify that "something" with a Matcher:
+
+> Expectation: `expect(dog).to `
+
+> Matcher: `be_a(Dog)`
+
+>We use `expect(IUT)` to "wrap" the ***Item Under Test***, so that it supports the `to` method which accepts a matcher. Here we are wrapping an object or block in expect, call to or to_not (aliased as not_to) and pass it a matcher object
+
+Now let's make the test **fail**. Let's say we "expect dog to be a Integer":
```
Comment (jshawl) [inline readme.md:406]: imagine people are gonna ask about other possible matchers here - include a link to documentation

File: readme.md [hunk 7]
```diff
@@ -304,318 +410,227 @@ Whenever a test fails, RSpec will put an error message after that test which sho
...
-  it "has the class Apartment" do
-    apartment = Apartment.new
-    expect(apartment).to be_a(Apartment)
-  end
-  it "has a String for an address" do
-    white_house = Apartment.new(address: "1600 Penn Ave")
-    expect(white_house.address).to be_a(String)
+  it "has a String for an Name" do
+    dog = Dog.new("Rover", 10)
+    expect(dog.name).to be_a(String)
```
Comment (jshawl) [inline readme.md]: recommend a repo with solution code for this example

File: readme.md [hunk 8]
```diff
@@ -304,318 +410,227 @@ Whenever a test fails, RSpec will put an error message after that test which sho
...
-describe Apartment do
-  describe "#add_tenant" do
-    context "when the number of tenants" do
-      context "is less than the number of beds" do
-        it "adds a tenant"
-      end
-      context "is equal to the number of beds" do
-        it "does not add a tenant"
-      end
+describe "#set_hunger_level" do
```
Comment (jshawl) [inline readme.md:457]: what's the naming convention here? when to use `#`? when to use `.`?

File: readme.md [hunk 9]
```diff
@@ -304,318 +410,227 @@ Whenever a test fails, RSpec will put an error message after that test which sho
...
-
-Crazy, right? Breathe, and let's go through it a bit at a time.
+Let's go through it a bit at a time:
 
 You can **nest** as many `describe` blocks as you want. What's the purpose? Just to lump together some tests. **It doesn't affect the code at all.** It's purely to keep things organized visually.
 
 Also, **`context` does literally the exact same thing as `describe`**. They're identical. RSpec makes no difference between them. So why have both? To make your tests more readable from an English standpoint. You can see here I'm using `describe` for when I'm talking about specific objects or methods, and `context` when I'm talking about, well, different contexts. If I replace `describe` with `context`, and vice-versa, the tests will run the exact same way.
 
-**RSpec is all about making tests easy to read from an English standpoint.**
+The hash `#` in front of `set_hunger_level` also doesn't do anything -- it's just what programmers usually use to indicate that something is a method, in the same way they use `$` to indicate a command you should enter in the terminal.
```
Comment (jshawl) [inline readme.md:473]: oops here it is

File: readme.md [hunk 10]
```diff
@@ -304,318 +410,227 @@ Whenever a test fails, RSpec will put an error message after that test which sho
...
 
-### TDD
+## You-Do: Purrspec (20 min)
 
-You've probably already run into the situation of not knowing when an app is "done". You've also probably started an assignment or project only to find there are an overwhelming number of things to do. Then, on top of that, you have all that by-hand, manual testing to do as you make your app.
+**Instructions:**
 
-When you write tests first, you're creating a tidy little checklist for yourself of things to complete. The **goal of unit tests** is that **when all of the tests pass, your app is complete**.
+1. Fork and Clone the following repo:
+[Purrspec](https://github.com/ga-dc/purrspec)
```
Comment (jshawl) [inline readme.md]: this is very similar to dog. can you think of another example that's a little closer to industry? maybe a shopping cart class with tests? I have bandwidth to help if you want

File: readme.md [hunk 11]
```diff
@@ -1,103 +1,201 @@
...
+* A little more time upfront means a lot of time saved down the line! (Think about refactoring)
+> Real World Example: DC Tech Startups
+
+<!--Planning on talking about two startups interviewed with and how wished they would have integrated more testing earlier on when developing their applications, it's now a big focus for them, cut corners early on and now need to refactor code base -->
+
+**DrawBacks**
+
+* Requires time and effort.
+
+* Could me more costly to an organization when there are changes in requirements.
```
Comment (joe-gz) [inline readme.md]: small typo: "could BE"

File: readme.md [hunk 12]
```diff
@@ -625,85 +640,31 @@ Your goal: When all the tests pass, that mean the robot works. However, you're o
...
+
+### Additional Resources
+- [Learn Ruby via RSpec]( https://github.com/ga-dc/learn_ruby_via_rspec)
+- [Code School RSpec](https://www.codeschool.com/courses/testing-with-rspec)
+- [RSpec for Newbies](http://code.tutsplus.com/tutorials/ruby-for-newbies-testing-with-rspec--net-21297)
+- Mocks & Stubs]
+- https://www.relishapp.com/rspec/rspec-mocks/docs
+- http://www.martinfowler.com/bliki/TestDouble.html
+- RSpec Cheatsheets:
+- https://www.anchor.com.au/wp-content/uploads/rspec_cheatsheet_attributed.pdf
```
Comment (jshawl) [inline readme.md:712]: additional exercises?

**Comment by amaseda** [comment]
Potential discussion point or a question to be prepared for is what entails too much testing. Is it possible to be too meticulous with your unit tests? Don't really have an answer to this but it's something I've always asked myself during testing classes.

**Pull request merged**